test/e2e/kappcontroller/package_repo_test.go

```diff
@@ -287,6 +288,414 @@ spec:
 	}
 }
 
+func Test_PackageReposWithOverlappingPackages_identicalPackagesReconcile(t *testing.T) {
```

Comment:
lets also add a test where same pkgr is added in global packaging namespace and in local one.